app/pulp/app/models/progress.py

```diff
@@ -70,14 +78,26 @@ def save(self, *args, **kwargs):
         """
         if self.task_id is None:
             self.task_id = Task.objects.get(id=util.get_current_task_id())
-        super(ProgressReport, self).save(*args, **kwargs)
+
+        now = timezone.now()
+
+
+        if self._using_context_manager and self._last_save_time:
+            if now - self._last_save_time >= datetime.timedelta(milliseconds=BATCH_INTERVAL):
+                super(ProgressReport, self).save(*args, **kwargs)
+        else:
+            super(ProgressReport, self).save(*args, **kwargs)
+
+        self._last_save_time = now
 
     def __enter__(self):
         """
         Saves the progress report state as RUNNING
         """
         self.state = self.RUNNING
         self.save()
+        self._using_context_manager = True
+        return self
```

Comment:
Oh actually we already claim that we do this so you are fixing it something for sure :)

https://github.com/werwty/pulp/blob/74755293cf5165db8a41f0d2ff7c4fd5482afe1d/app/pulp/app/models/progress.py#L206

I don't think any docs changes are necessary then.